src/PSR7/ServerRequestValidator.php

```diff
@@ -96,251 +89,17 @@ public function validate(ServerRequestInterface $serverRequest) : OperationAddre
         throw MultipleOperationsMismatchForRequest::fromMatchedAddrs($matchingOperationsAddrs);
     }
 
-    protected function validateAddress(OperationAddress $addr, ServerRequestInterface $serverRequest) : void
-    {
-        // 1. Headers
-        $this->validateHeaders($addr, $serverRequest);
-
-        // 2. Cookies
-        $this->validateCookies($addr, $serverRequest);
-
-        // 3. Body
-        $this->validateBody($addr, $serverRequest);
-
-        // 4. Validate Query arguments
-        $this->validateQueryArgs($addr, $serverRequest);
-
-        // 5. Validate path
-        $this->validatePath($addr, $serverRequest);
-
-        // 6. Validate security
-        $this->validateSecurity($addr, $serverRequest);
-    }
-
     /**
-     * @throws TypeErrorException
+     * Check the openapi spec and find matching operations(path+method)
+     * This should consider path parameters as well
+     * "/users/12" should match both ["/users/{id}", "/users/{group}"]
+     *
+     * @return OperationAddress[]
      */
-    protected function validateHeaders(OperationAddress $addr, ServerRequestInterface $serverRequest) : void
+    private function findMatchingOperations(ServerRequestInterface $request) : array
```

Comment:
I found this function was used privately by this class, so moved it here from specfinder, since it is not really specs, but path matching (aka routing)